### DISCUSSION TITLE
[Help] [Node.js] [Wireless] termux websockify.js startup error

### DISCUSSION THREAD

**Discussion by CorruptionVR** [yume-chan/ya-webadb#624] [Q&A] [answered]
ok, so I'm using Termux on my Android and I'm using the command
```shell
node websockify.js 10.7.25.176:33529 :15555 --key self.pem --cert self.pem
```
from `~/websockify-js/websockify` and im getting this error
```shell
WebSocket settings:
    - proxying from 10.7.25.176:33529 to :15555
    - Running in encrypted HTTPS (wss://) mode using: self.pem, self.pem
node:events:489
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::33529
    at Server.setupListenHandle [as _listen2] (node:net:1823:16)
    at listenInCluster (node:net:1871:12)
    at Server.listen (node:net:1959:7)
    at Object.<anonymous> (/data/data/com.termux/files/home/websockify-js/websockify/websockify.js:222:11)
    at Module._compile (node:internal/modules/cjs/loader:1255:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1309:10)
    at Module.load (node:internal/modules/cjs/loader:1113:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1850:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -98,
  syscall: 'listen',
  address: '::',
  port: 33529
}

```
I made self.pem using OpenSSL
what's going wrong and how can I get it to work?

**Comment by yume-chan** [comment]
The usage is `node websockify.js <local-port> <remote-address>`. It listens on `<local-port>` for WebSocket connections, and forwards them to `<remote-address>`.

So it should be `node websocket.js 15555 localhost:33529`. Then connect to `wss://<device-ip>:15555` in old demo.

WebSockify is really difficult to use so I don't recommend.

**Reply by CorruptionVR** [reply]
is the output supposed to end at ` - Running in unencrypted HTTP (ws://) mode`, and not return you to being able to use the terminal?

**Reply by yume-chan** [reply]
> is the output supposed to end at ` - Running in unencrypted HTTP (ws://) mode`

No, ` --key self.pem --cert self.pem` is still required to run in wss mode. `- Running in encrypted HTTPS (wss://) mode using: self.pem, self.pem` from your original post is correct. Remember to trust the certificate on PC.

> and not return you to being able to use the terminal?

Yes, it won't exit. I forgot how to create a new session in Termux, but you should be able to use tmux to create multiple sessions https://tmuxcheatsheet.com/

**Reply by CorruptionVR** [reply]
for the "is the output supposed to end at ` - Running in unencrypted HTTP (ws://) mode`", I should have specified that I tested it without the cetificate

**Reply by yume-chan** [reply]
It will stop there until a WebSocket connection is made to it.

**Comment by CorruptionVR** [comment, marked as answer]
Wow, I just found this, and i got it working instantly
https://github.com/nwtgck/piping-adb-web
gonna try to make a shell script to automate the setup

edit: never mind. the command just works when i turn on wireless debugging
